src/on.js

```diff
@@ -6,171 +6,229 @@
+					on(node, 'DOMMouseScroll', callback),
+					on(node, 'mousewheel', callback)
+				]);
 			}
-		};
+		}
+
+		// special case: keyboard
+		if (/^key/.test(eventName)) {
+			callback = normalizeKeyEvent(callback);
+		}
+
+		// default case
+		return on.onDomEvent(node, eventName, callback);
 	}
 
-	function onClickoff (node, callback) {
-		// important note!
-		// starts paused
-		//
-		var
-			handle,
-			bHandle = on(document.body, 'click', function (event) {
-				var target = event.target;
+	// registered functional events
+	on.events = {
+		// handle click and Enter
+		button: function (node, callback) {
+			return on.makeMultiHandle([
+				on(node, 'click', callback),
+				on(node, 'keyup:Enter', callback)
+			]);
+		},
+
+		// custom - used for popups 'n stuff
+		clickoff: function (node, callback) {
+			// important note!
+			// starts paused
+			//
+			var bHandle = on(node.ownerDocument.documentElement, 'click', function (e) {
+				var target = e.target;
 				if (target.nodeType !== 1) {
 					target = target.parentNode;
 				}
 				if (target && !node.contains(target)) {
-					callback(event);
+					callback(e);
 				}
 			});
 
-		handle = {
-			state: 'resumed',
-			resume: function () {
-				setTimeout(function () {
-					bHandle.resume();
-				}, 100);
-				this.state = 'resumed';
+			var handle = {
+				state: 'resumed',
+				resume: function () {
+					setTimeout(function () {
+						bHandle.resume();
+					}, 100);
+					this.state = 'resumed';
+				},
+				pause: function () {
+					bHandle.pause();
+					this.state = 'paused';
+				},
+				remove: function () {
+					bHandle.remove();
+					this.state = 'removed';
+				}
+			};
+			handle.pause();
+
+			return handle;
+		}
+	};
+
+	// internal event handlers
+
+	function onDomEvent (node, eventName, callback) {
+		node.addEventListener(eventName, callback, false);
+		return {
+			remove: function () {
+				node.removeEventListener(eventName, callback, false);
+				node = callback = null;
+				this.remove = this.pause = this.resume = function () {};
 			},
 			pause: function () {
-				bHandle.pause();
-				this.state = 'paused';
+				node.removeEventListener(eventName, callback, false);
 			},
-			remove: function () {
-				bHandle.remove();
-				this.state = 'removed';
+			resume: function () {
+				node.addEventListener(eventName, callback, false);
 			}
 		};
+	}
 
-		handle.pause();
+	function onImageLoad (node, callback) {
+		var handle = on.makeMultiHandle([
+			on.onDomEvent(node, 'load', onImageLoad),
+			on(node, 'error', callback)
+		]);
 
 		return handle;
-	}
 
-	function onImageLoad (img, callback) {
 		function onImageLoad (e) {
-			var h = setInterval(function () {
-				if (img.naturalWidth) {
-					e.width = img.naturalWidth;
-					e.naturalWidth = img.naturalWidth;
-					e.height = img.naturalHeight;
-					e.naturalHeight = img.naturalHeight;
+			var interval = setInterval(function () {
+				// TODO: what if an image is 0x0? is it possible?
```

Comment:
I think it is a rare case, if at all possible. I suggest to skip this corner case for now.